File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        table = {}
        for i, val in enumerate(nums):
            if val not in table.keys():
                diff = target - val
                table[diff] = i
            else:
                return [table[val], i]

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    prev = {}
    
    for (let i = 0; i < nums.length; i++){
        let diff = target - nums[i];
        
        if (diff in prev){
            return [prev[diff], i];
        }
        
        prev[nums[i]] = i;
        
    }
};